snapshots/output-net6.0/syntax/Main/Methods.cs

```diff
@@ -210,4 +210,35 @@ public static void InheritedOverloads()
 //                                  ^^^^^^ reference scip-dotnet nuget . . Main/Methods#InheritedOverloads3#Method().
       }
 
+      public class LocalFunction
+//                 ^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Methods#LocalFunction#
+//                               documentation ```cs\nclass LocalFunction\n```
+      {
+          public static void Method()
+//                           ^^^^^^ definition scip-dotnet nuget . . Main/Methods#LocalFunction#Method().
+//                                  documentation ```cs\npublic static void LocalFunction.Method()\n```
+          {
+              var myWorld = GetWorld();
+//                ^^^^^^^ definition local 0
+//                        documentation ```cs\nstring? myWorld\n```
+//                          ^^^^^^^^ reference scip-dotnet nuget . . Main/Methods#LocalFunction#Method().GetWorld().
+              SayHi(myWorld);
+//            ^^^^^ reference scip-dotnet nuget . . Main/Methods#LocalFunction#Method().SayHi().
+//                  ^^^^^^^ reference local 0
+
+              string GetWorld() => "world";
+//                   ^^^^^^^^ definition scip-dotnet nuget . . Main/Methods#LocalFunction#Method().GetWorld().
```

Comment:
Can local functions be referenced outside of the function? If not, then they should be treated as local symbols. Local symbols are generally preferable over global symbols because they enable cheaper indexing ("find refs" only needs to look at the current document)

Reply:
They can be only used inside the member they are defined in, so yes it makes sense to make them local.
On the same note, I think we should make the method parameters local symbols too. WDYT?

Reply:
If method parameters can't be accessed outside the document then they should be local as well.

FWIW, we recently added new fields to `SymbolInformation` to close the gap between local symbols and global symbols so you don't make a tradeoff wrt. to functionality when choosing to use a local symbol over a global symbol (most notably, `display_name`, `enclosing_symbol`). Not something we need to care about in scip-dotnet for now, just an FYI.

Reply:
I think this one can be merged, I will take a look at the method parameters in a different PR at some point later. The parameters are only visible inside the method body. So, I think it makes sense to make them local.

Reply:
Great! Already merged.